worker/uniter/operation/runhook.go

```diff
@@ -251,6 +247,12 @@ func (rh *runHook) Commit(state State) (*State, error) {
 			Step: Queued,
 			Hook: hi,
 		}
+	case hooks.PreSeriesUpgrade:
+		message := createUpgradeSeriesStatusMessage(rh.name, rh.hookFound)
+		err = rh.callbacks.SetUpgradeSeriesStatus(model.UpgradeSeriesPrepareCompleted, message)
+	case hooks.PostSeriesUpgrade:
+		message := createUpgradeSeriesStatusMessage(rh.name, rh.hookFound)
+		err = rh.callbacks.SetUpgradeSeriesStatus(model.UpgradeSeriesCompleted, message)
```

Comment:
Here we move the updates down to the commit phase and as close to the end of it as possible.